src/pyth_utils.rs

```diff
@@ -0,0 +1,321 @@
+use std::path::PathBuf;
+
+use base64::{
+    engine::general_purpose,
+    prelude::{Engine, BASE64_STANDARD},
+};
+use fuels::{
+    prelude::{abigen, CallParameters, Contract, LoadConfiguration, TxPolicies, WalletUnlocked},
+    programs::call_response::FuelCallResponse,
+    types::{Address, Bits256, Bytes, Identity},
+};
+use rand::Rng;
+
+abigen!(Contract(
+    name = "PythOracleContract",
+    abi = "pyth-contract/out/debug/pyth-contract-abi.json"
+));
+
+pub struct Pyth {
+    pub instance: PythOracleContract<WalletUnlocked>,
+    pub wallet: WalletUnlocked,
+}
+
+const ORACLE_CONTRACT_BINARY_PATH: &str = "./pyth-contract/out/debug/pyth-contract.bin";
```

Comment:
The remaining constants should be added to the constants file, and test values should be distinguished

Reply:
100%